### ISSUE TITLE
Plans: one click purchase success page

### ISSUE THREAD

**Issue by gwwar** [Automattic/wp-calypso#7615]
Note that we should wait until #7614 is done before starting this task.

As part of #7329 if we have a one-click checkout flow, the success page (maybe the thank-you page) will need to give context as to what has happened, and offer a refund link if this purchase was made in error.

**Comment by folletto** [comment]
> As part of #7329 if we have a one-click checkout flow, the success page (maybe the thank-you page) will need to give context as to what has happened, and offer a refund link if this purchase was made in error.

I'm commenting given the "Design Review" tag on this issue → it's a bit hard to get the full context on this. Is a design required? For what? What's the full flow of this? 

This seems a thing that has to be designed in full first (P2?), before breaking down in multiple GitHub issues.

**Comment by gwwar** [comment]
> This seems a thing that has to be designed in full first (P2?), before breaking down in multiple GitHub issues.

@folletto Part of the problem is that we don't have design resources at the moment. The main issue is located at #7329 and p3fqKv-3md-p2

#7329 intends to add a one-click purchase flow. This is a placeholder task to give users a chance to cancel this purchase if they clicked the button in error.

**Comment by apeatling** [comment]
@gwwar I think we should be using the exact same "thank you, and here's your plan" page that we use when someone has bought a plan in the NUX. Once they click that button, send them there with confirmation of the purchase.

I don't think we should be making it any easier to refund that it already is, they can still cancel using the regular "manage purchases" UI.

**Comment by gwwar** [comment]
> I don't think we should be making it any easier to refund than it already is, they can still cancel using the regular "manage purchases" UI.

I don't think we should put this on the thank-you page for all everyone. This should show up for just the people who went through the one-click purchase. It can be as simple as a notice that links to the manage purchases page. I think it's important here because folks might click on this by accident.

**Comment by apeatling** [comment]
I think I would prefer a confirmation prompt than a prominent refund link.

**Comment by gwwar** [comment]
> I think I would prefer a confirmation prompt than a prominent refund link.

@apeatling Do you mind describing that in #7329? Is this a modal prompt or do we simply go to checkout?

**Comment by apeatling** [comment]
@gwwar I've thought about this some more. I think having a link to the "manage purchases" section is better, but we should be careful with the wording. Just a "Manage your purchases" link would work, so avoiding anything loaded like "Get a refund".

**Comment by folletto** [comment]
> Part of the problem is that we don't have design resources at the moment. 

Noted. That however to me makes it even more important to have the discussion in a single, clear post on a P2, where more designer can chime in and produce a single flow, than a few GitHub issues that requires some work to connect them together.

The discussion just happened above is evidence of the need of such a consolidated thread imho. :)

That said, if you feel more comfortable here it's no problem. We just need to involve more designers if there's no designer on team (direct pings? discussion on slack? start a P2 thread now?).

**Comment by melchoyce** [comment]
> Noted. That however to me makes it even more important to have the discussion in a single, clear post on a P2, where more designer can chime in and produce a single flow, than a few GitHub issues that requires some work to connect them together.

I agree with @folletto — it sounds like it would be good to take a step back and walk through the flow together, then make a flowchart document. That kind of thing you won't need a designer for. :)

**Comment by gwwar** [comment]
Thanks @folletto and @melchoyce! It looks like the flow has simplified a bit (correct me if you think otherwise) so I've added an annotated mock to the master thread in p3fqKv-3md-p2. 

With the new plan purchase details that will be added in https://github.com/Automattic/wp-calypso/pull/5212 almost in, it should be safe to close out this task. We can add a similar module to the thank you page as we do in my-plans.
